src/main/scala/com/research/nomad/markii/dataflow/AbstractValuePropVASCO.scala

```diff
@@ -324,12 +397,32 @@ class AbstractValuePropVASCO(entryPoints: List[SootMethod])
     d.setHandlers(context.getMethod, stmt, viewBase, eventType, handlers)
   }
 
+  /**
+   * Add handler to the call-graph and abstract state
+   */
+  def setEventHandler(d: AFTDomain, windowClass: SootClass,
+                      handler: SootMethod, eventType: EventType, viewNode: ViewNode): AFTDomain = {
+    // FIXME: windowClass vs Activity? Maybe we can unify them?....since the virtual/fake method contains the constructed
+    //  instance for all em'
+    DynamicCFG.addViewHandlerToEventLoopAct(windowClass, handler) match {
+      case Some((runner, invocation)) =>
+        aftProgramRepresentation.refreshCFGcache(runner)
+        for (runnerContext <- getContexts(runner).asScala) {
+          runnerContext.setValueBefore(invocation, topValue())
+          runnerContext.setValueAfter(invocation, topValue())
+        }
+        initContext(runner, topValue())
+      case None =>
+    }
+    d.addHandler(viewNode, eventType, handler)
+  }
+
   override def callLocalFlowFunction(context: DomainContext, unit: soot.Unit, d: Domain): Domain = {
     logState(context.getMethod, unit, d)
```

Comment:
@LuxiWang99 To test this PR, you can try to pull down this PR's code and set a breakpoint here with `context.getMethod.getName == "fire"` (details might be different, but you got the idea)